### PR TITLE
Add spiral-matrix

### LINKED ISSUE #752: spiral-matrix: implement exercise
There are exercises that are not yet implemented in Python track (the full list of them - https://github.com/exercism/python/issues/417#issuecomment-334841778).
You can help us change this situation by implementing this one. Ref #417

The guide ["Implement an exercise from specification"](https://github.com/exercism/docs/blob/master/you-can-help/implement-an-exercise-from-specification.md) will help you succeed.

If you still have questions, I encourage you to ask them before starting working on the issue.

I also suggest claiming the issue by creating an empty PR as stated [here](https://github.com/exercism/docs/blob/master/you-can-help/implement-an-exercise-from-specification.md#avoiding-duplicate-work)

Please note that there are some special ['closing issue' keywords](https://help.github.com/articles/closing-issues-using-keywords/) that you can put in description of your PR and Github will take care of closing the corresponding issue automatically after merge.
This simple thing helps us a lot, and makes our life a bit easier

### PR DESCRIPTION
**Pull request by chgraef**
Implementing exercise spiral-matrix. Closes #752

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -1268,6 +1268,19 @@
...
+        "uuid": "b0c7cf95-6470-4c1a-8eaa-6775310926a2",
+        "slug": "spiral-matrix",
+        "core": false,
+        "unlocked_by": null,
+        "difficulty": 2,
+        "topics": [
+            "algorithms",
+            "control-flow",
+            "lists",
+            "logic"
```
Comment (cmccandless) [inline config.json]: I don't think that `logic` really applies to this exercise. A better example would be the [`zebra-puzzle`](https://github.com/exercism/python/tree/master/exercises/zebra-puzzle) exercise.

File: exercises/spiral-matrix/spiral_matrix_test.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import unittest
+
+from spiral_matrix import spiral
+
+
```
Comment (cmccandless) [inline exercises/spiral-matrix/spiral_matrix_test.py]: Please add a comment referencing the current version of the [canonical data](https://github.com/exercism/problem-specifications/blob/master/exercises/spiral-matrix/canonical-data.json). For the format of this comment, please see #784.

File: config.json (hunk 1 above)
Comment (chgraef) [reply config.json]: Good point, @cmccandless. I decided to delete "logic" from the list of topics.

File: exercises/spiral-matrix/spiral_matrix_test.py (hunk 2 above)
Comment (chgraef) [reply exercises/spiral-matrix/spiral_matrix_test.py]: Done. Thanks for the hint.

**Comment by cmccandless** [comment]
@chgraef Please add a reference to #752 to the PR description (see [magic words](https://help.github.com/articles/closing-issues-using-keywords/)).

File: exercises/spiral-matrix/spiral_matrix_test.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import unittest
+
+from spiral_matrix import spiral
+
+# Tests adapted from `problem-specifications//canonical-data.json` @ v1.1.0
```
Comment (N-Parsons) [inline exercises/spiral-matrix/spiral_matrix_test.py]: Can you move this line down by one to be more consistent with the other test files?

Comment (chgraef) [reply exercises/spiral-matrix/spiral_matrix_test.py]: Thanks, @N-Parsons ,will do.

Comment (cmccandless) [reply exercises/spiral-matrix/spiral_matrix_test.py]: I believe for full consistency, you the spacing should be 2 blank lines above, one below (see below):

```Python
from spiral_matrix...


# Tests adapted from...

class...

**Comment by cmccandless** [comment]
@chgraef Merged; thanks for working on this!

**Pull request merged**